build-system/tasks/helpers.js

```diff
@@ -285,57 +285,91 @@ function appendToCompiledFile(srcFilename, destFilePath) {
+  const startTime = Date.now();
   const entryPoint = path.join(srcDir, srcFilename);
+  return closureCompile(entryPoint, destDir, options.minifiedName, options)
+      .then(function() {
+        const destPath = path.join(destDir, options.minifiedName);
+        appendToCompiledFile(srcFilename, destPath);
+        fs.writeFileSync(
+            path.join(destDir, 'version.txt'), internalRuntimeVersion);
+        if (options.latestName) {
+          fs.copySync(
+              destPath,
+              path.join(destDir, options.latestName));
+        }
+      })
+      .then(() => {
+        let name = options.minifiedName;
+        if (options.latestName) {
+          name = `${name} → ${options.latestName}`;
+        }
+        endBuildStep('Minified', name, startTime);
+      });
+}
 
-  // Transpile TS to Closure-annotated JS before actual bundling or compile.
-  if (options.typeScript) {
-    const startTime = Date.now();
-    transpileTs(srcDir, srcFilename);
-    endBuildStep('Transpiled', srcFilename, startTime);
+/**
+ * Handles a browserify bundling error
+ * @param {Error} err
+ * @param {boolean} failOnError
+ * @param {string} srcFilename
+ * @param {string} startTime
+ */
+function handleBundleError(err, failOnError, srcFilename, startTime) {
+  let message = err;
+  if (err.stack) {
+    // Drop the node_modules call stack, which begins with '    at'.
+    message = err.stack.replace(/    at[^]*/, '').trim();
   }
+  console.error(red(message));
+  if (failOnError) {
+    process.exit(1);
+  } else {
+    endBuildStep('Error while compiling', srcFilename, startTime);
+  }
+}
 
-  if (options.minify) {
-    const startTime = Date.now();
-    return closureCompile(entryPoint, destDir, options.minifiedName, options)
-        .then(function() {
-          const destPath = path.join(destDir, options.minifiedName);
-          appendToCompiledFile(srcFilename, destPath);
-          fs.writeFileSync(
-              path.join(destDir, 'version.txt'), internalRuntimeVersion);
-          if (options.latestName) {
-            fs.copySync(
-                destPath,
-                path.join(destDir, options.latestName));
-          }
-        })
-        .then(() => {
-          let name = options.minifiedName;
-          if (options.latestName) {
-            name = `${name} → ${options.latestName}`;
-          }
-          endBuildStep('Minified', name, startTime);
-
-          // Remove intemediary, transpiled JS files after compilation.
-          if (options.typeScript) {
-            rimraf.sync(path.join(srcDir, '**/*.js'));
-          }
-        });
+/**
+ * Performs the final steps after Browserify bundles a JS file
+ * @param {string} srcFilename
+ * @param {string} destDir
+ * @param {string} destFilename
+ * @param {?Object} options
+ */
+function finishBundle(srcFilename, destDir, destFilename, options) {
+  appendToCompiledFile(srcFilename,
+      path.join(destDir, destFilename));
+
+  if (options.latestName) {
```

Comment:
if options is actually nullable, we should do `if (options && options.latestName) {`

Reply:
Same comment as above. Bug fixes / rewrites will happen in a follow up.